core/src/main/kotlin/io/specmatic/core/pattern/QueryParameterScalarPattern.kt

```diff
@@ -17,23 +17,17 @@ data class QueryParameterScalarPattern(override val pattern: Pattern): Pattern b
     }
 
     override fun matches(sampleData: Value?, resolver: Resolver): Result {
-        if(sampleData == null){
-            return Result.Failure("Null found, expected a scalar value")
-        }
+        if (sampleData == null) return dataTypeMismatchResult(pattern, sampleData, resolver.mismatchMessages)
         val sampleDataString = when (sampleData) {
             is ListValue -> {
-                if (sampleData.list.size > 1) return Result.Failure("Multiple values $sampleData found. Expected a single value")
+                if (sampleData.list.size > 1) return dataTypeMismatchResult(pattern, sampleData, resolver.mismatchMessages)
                 sampleData.list.single().toStringLiteral()
             }
             else -> sampleData.toStringLiteral()
         }
 
-        return try {
-            val parsedValue = runCatching { pattern.parse(sampleDataString, resolver) }.getOrDefault(StringValue(sampleDataString))
-            resolver.matchesPattern(null, pattern, parsedValue)
-        } catch (e: Throwable) {
-            Result.Failure(exceptionCauseMessage(e))
-        }
+        val parsedValue = runCatching { pattern.parse(sampleDataString, resolver) }.getOrDefault(StringValue(sampleDataString))
+        return resolver.matchesPattern(null, pattern, parsedValue)
```

Comment:
is it possible for matchesPattern to throw an exception? If it does, it will bubble up as an exception, rather than a failure, since the try-catch has been removed.